cookbooks/cdo-ruby/recipes/2.0.rb

```diff
@@ -43,17 +39,13 @@
   action :nothing
 end
 
-gem_package "bundler" do
-  action :upgrade
-  version '1.10.4'
+# Fix RubyGems bug that created ~/.gem with root permissions in the underlying user's home folder.
+require 'etc'
+user = Etc.getlogin
```

Comment:
This is the future-proof way to retrieve the underlying logged-in user from within a `sudo chef-client` run (`node[:current_user]`, which is used throughout our other cookbooks, no longer works in Chef >= 12, since `current_user` was changed to return `root` in Chef >= 12). I plan to submit a separate PR to update the rest of our cookbooks with this fix at some point.